log4j-core/src/main/java/org/apache/logging/log4j/core/impl/ThrowableProxy.java

```diff
@@ -401,15 +423,17 @@ public String getExtendedStackTraceAsString(
      * @param textRenderer The message renderer.
      * @param suffix Append this to the end of each stack frame.
      * @param lineSeparator The end-of-line separator.
+     * @param maxLineCount The total line count of final result
      */
     public void formatExtendedStackTraceTo(
             final StringBuilder sb,
             final List<String> ignorePackages,
             final TextRenderer textRenderer,
             final String suffix,
-            final String lineSeparator) {
+            final String lineSeparator,
+            final Integer maxLineCount) {
```

Comment:
…this is causing a breaking change.

You need to **deprecate**, but keep the 5-param version of `formatExtendedStackTraceTo`:

```java
  /**
   * @deprecated since 2.24.0
   */
   @Deprecated
    public void formatExtendedStackTraceTo(
            final StringBuilder sb,
            final List<String> ignorePackages,
            final TextRenderer textRenderer,
            final String suffix,
            final String lineSeparator) {
        formatExtendedStackTraceTo(sb, ignorePackages, textRenderer, suffix, lineSeparator, null);
    }
```
and then…

Reply:
Oh...I understand it now. Thanks for the guidance!